c7n/resources/s3.py

```diff
@@ -1389,19 +1390,19 @@ class FilterPublicBlock(Filter):
                 region: us-east-1
                 filters:
                   - type: check-public-block
-                    scope: Any
-                    enabled: False
+                  # BlockPublicAcls: true <-- optional, if none provided sets all to true by default
 
     """
 
     schema = type_schema(
         'check-public-block',
-        scope={
-            'type': 'string',
-            'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
-                'BlockPublicPolicy', 'RestrictPublicBuckets', 'All', 'Any']},
-        enabled={'type': 'boolean'})
+        BlockPublicAcls={'type': 'boolean'},
+        IgnorePublicAcls={'type': 'boolean'},
+        BlockPublicPolicy={'type': 'boolean'},
+        RestrictPublicBuckets={'type': 'boolean'})
     permissions = ("s3:GetBucketPublicAccessBlock",)
+    keys = (
+        'BlockPublicPolicy', 'BlockPublicAcls', 'IgnorePublicAcls', 'RestrictPublicBuckets')
 
     def process(self, buckets, event=None):
         results = list()
```

Comment:
for builtin types use the literal syntax instead of type as constructor, ie `results = []`

Reply:
in general premature optimizations... generally should be avoided. optimizations should be profiler guided or hypothesis tested as useful. a single network api call, renders these types of micro things silly. general goal is to strive for correctness and maintainability (obviousness).